### PR TITLE
rxlifecycle-kotlin module added with bind() syntax

### PR DESCRIPTION
**Pull request by dsvoronin**
address reverted #58 , and some nice new syntax

Not sure if any tests will work here.

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -1,4 +1,4 @@
-include ':rxlifecycle'
+include ':rxlifecycle', ':rxlifecycle-kotlin'
```
Comment (dlew) [inline settings.gradle]: Could this be on a newline to match the rest of it?

File: rxlifecycle-kotlin/src/main/java/com/trello/rxlifecycle/kotlin/rxlifecycle.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import com.trello.rxlifecycle.ActivityLifecycleProvider
+import com.trello.rxlifecycle.FragmentLifecycleProvider
+import com.trello.rxlifecycle.RxLifecycle
+import rx.Observable
+
+fun <T> Observable<T>.bind(activity: ActivityLifecycleProvider): Observable<T> = this.compose<T>(activity.bindToLifecycle<T>())
+
+fun <T> Observable<T>.bind(fragment: FragmentLifecycleProvider): Observable<T> = this.compose<T>(fragment.bindToLifecycle<T>())
+
+fun <T> Observable<T>.bind(view: View): Observable<T> = this.compose<T>(RxLifecycle.bindView(view))
```
Comment (dlew) [inline rxlifecycle-kotlin/src/main/java/com/trello/rxlifecycle/kotlin/rxlifecycle.kt]: There should be both `bindToLifecycle()` and `bindUntilEvent()` for these.

File: rxlifecycle-kotlin/src/main/java/com/trello/rxlifecycle/kotlin/rxlifecycle.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.trello.rxlifecycle.kotlin
```
Comment (dlew) [inline rxlifecycle-kotlin/src/main/java/com/trello/rxlifecycle/kotlin/rxlifecycle.kt]: Should have copyright at top of file.

File: rxlifecycle-kotlin/gradle.properties [hunk 4]
```diff
@@ -0,0 +1,4 @@
+POM_NAME=RxLifecycle-Kotlin
+POM_DESCRIPTION=RxLifecycle-Kotlin
+POM_ARTIFACT_ID=rxlifecycle-Kotlin
```
Comment (dlew) [inline rxlifecycle-kotlin/gradle.properties]: No capital 'k' here.

File: settings.gradle (hunk 1 above)
Comment (dsvoronin) [inline settings.gradle]: of course!
And since kotlin just released, i have to upgrade it

File: rxlifecycle-kotlin/build.gradle [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    compile kotlinStdlib
+    compile project(':rxlifecycle')
+
+    compile rootProject.ext.rxJava
+
+    compile rootProject.ext.appCompat
+
+    testCompile rootProject.ext.junit
+    testCompile rootProject.ext.mockito
+    testCompile rootProject.ext.robolectric
```
Comment (dlew) [inline rxlifecycle-kotlin/build.gradle]: Does this part of the project really need all these dependencies? I'm guessing it only needs the top two.

File: build.gradle [hunk 6]
```diff
@@ -1,9 +1,15 @@
 buildscript {
+    ext {
+        //version here to share between build script and projects
+        //todo https://github.com/JetBrains/kotlin/pull/775 : remove if it will be merged
+        verKotlin = '1.0.0-rc-1036'
```
Comment (dlew) [inline build.gradle]: Kotlin 1.0 final got released, should update here.

File: README.md [hunk 7]
```diff
@@ -70,6 +70,15 @@ public class MyActivity extends NaviActivity {
 }
 ```
 
+If you want some kotlin goodness, you can use bind() syntax
+
+```java
+myObservable
+    .bind(myView) //bind(myRxActivity or myRxFragment)
+    .subscribe()
+
```
Comment (dlew) [inline README.md]: Exatra newline here.

File: README.md [hunk 8]
```diff
@@ -70,6 +70,15 @@ public class MyActivity extends NaviActivity {
 }
 ```
 
+If you want some kotlin goodness, you can use bind() syntax
+
+```java
+myObservable
+    .bind(myView) //bind(myRxActivity or myRxFragment)
```
Comment (dlew) [inline README.md]: This comment seems unnecessary.

**Comment by dsvoronin** [comment]
all things solved!

File: README.md [hunk 9]
```diff
@@ -70,6 +70,22 @@ public class MyActivity extends NaviActivity {
...
+
+```java
+myObservable
+    .bindToLifecycle(myView)
+    .subscribe { }
+
+```
+
+```java
+myObservable
```
Comment (dlew) [inline README.md]: I would just make this a single example instead of two.

Comment (dlew) [inline README.md]: Eh, can deal with this post-merge.

**Comment by dlew** [comment]
Thanks!

File: README.md (hunk 9 above)
Comment (dsvoronin) [inline README.md]: You want me to create example android project?

Comment (dlew) [inline README.md]: Nah. I was just saying that this is two separate code samples but could just be wrapped into one.

**Pull request merged**